File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) {
        stack<char> st;
        int i=0; 
        while (i<s.length()){
            char c = s[i];

            if (c == '*'){
                if (!st.empty()){
                    st.pop();
                }
            } else {
                st.push(c);
            }
            i+=1;
        }
        string ans;
        while (!st.empty()){
            ans+=st.top();
            st.pop();
        }

        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.go
func removeStars(s string) string {
    var st []byte
    for i := range s {
        if s[i] == '*' {
            st = st[:len(st)-1]
        } else {
            st = append(st, s[i])
        }
    }

    return string(st)
}